### PR TITLE
Registration flow

### PR DESCRIPTION
**Pull request by zeroXten**
Its a bit of a mess still. I keep mentally flipping between the CLI being an offline tool and the CLI being a prototype for how the API will work. The idea is to support an offline mode even when the API works, so it will probably just be a nice wrapper around the API functions.

* Closed previous PR because it came from my master
* Interesting to note that gom install doesn't update what is already installed. This meant I was working with my old copy of the pki.io packages. I deleted _vendor and gom install'ed.

[Review on Reviewable](https://reviewable.io/reviews/pki-io/admin/41)

### CODE REVIEW

**Comment by thanasisk** [comment]
Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/pki-io/admin/41)

Reviewed files:
- .travis.yml @ r1
- Makefile @ r1





---

File: runOrg.go [hunk 1]
```diff
@@ -84,11 +84,28 @@ func orgRegisterNodes(argv map[string]interface{}) (err error) {
...
+			nodeContainer.Data.Body = node.Dump()
+			if err := org.Sign(nodeContainer); err != nil {
+				fsAPI.PushIncoming(fsAPI.Id, "registration", regJson)
+				panic(fmt.Sprintf("Couldn't sign node container: %s", err.Error()))
+
+			}
+			/*if err := fsAPI.PushIncoming(node.Data.Body.Id, "registration"); err != nil {
+				panic(fmt.Sprintf("Couldn't push node: %s", err.Error()))
+			}*/
+
```
Comment (jonbonazza) [inline runOrg.go]: I think we should try to remove commented code before committing it. Keeps things cleaner. :+1:

File: runPairingKey.go [hunk 2]
```diff
@@ -16,7 +16,11 @@ func pairingKeyNew(argv map[string]interface{}) (err error) {
 	index := LoadIndex(fsAPI, org)
 
 	id := NewID()
-	key := hex.EncodeToString(crypto.RandomBytes(16))
+	random, err := crypto.RandomBytes(16)
+	if err != nil {
+		panic(fmt.Sprintf("Couldn't get random bytes: %s", err.Error()))
```
Comment (jonbonazza) [inline runPairingKey.go]: The error  returned by RandomBytes() function already says basically the exact same thing. I think we may want to reconsider our conventions on where this kind of context is placed. 

Also, err.String() calls err.Error() and fmt.Sprintf() implicitly calls err.String, so there isn't a need to call err.Error(). Could just do:
panic(fmt.Sprintf("Couldn't get random bytes: %s", err))

**Comment by zeroXten** [comment]
Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/pki-io/admin/41)



---

Ah ok. Does fmt.Errorf also call String()? I remember seeing a lot of ugly stuff around the errors when I just used err directly.

---

**Comment by jonbonazza** [comment]
Ill have to test,  but i thought it did. Try it in the go playground and
see. :)

**Comment by jonbonazza** [comment]
Just tested. Yes it works.

**Comment by zeroXten** [comment]
Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/pki-io/admin/41)



---

Ah cool. Will using the logging library change significantly how we use errors anyway?

---

**Comment by jonbonazza** [comment]
Kind of. We will still use fmt.Error() and such to create error objects to
return,  but the end consumer app will use the seelog package instead of
the log package to write the log. Ill create a task to setup seelog for us
and replace the current logging statements with the seelog package. There
is an idiomatic way to do this,  so i'll assign the task to me to start. :)

**Comment by jonbonazza** [comment]
Went ahead and merged. Making some changes to this package anyway for logging, so I will just make the changes I mentioned. =)

**Pull request merged**